lib/ReactViews/ExplorerWindow/ExplorerWindow.jsx

```diff
@@ -33,15 +34,25 @@ const ExplorerWindow = createReactClass({
 
     /* eslint-disable-next-line camelcase */
     UNSAFE_componentWillMount() {
+        const { terria, viewState } = this.props;
+
         this.props.viewState.explorerPanelAnimating = true;
 
         this._pickedFeaturesSubscription = ko.pureComputed(this.isVisible, this).subscribe(this.onVisibilityChange);
+
+        if (terria.sharedFromExplorerPanel) {
+            viewState.openAddData();
+        }
+        if (defined(terria.previewedItemId) && terria.catalog.shareKeyIndex[terria.previewedItemId]) {
+            viewState.viewCatalogMember(terria.catalog.shareKeyIndex[terria.previewedItemId]);
+        }
         this.onVisibilityChange(this.isVisible());
     },
 
     componentDidMount() {
         this.escKeyListener = e => {
-            if (e.keyCode === 27) {
+            // Only explicitly check share modal state, move to levels/"layers of modals" logic if we need to go any deeper
+            if (e.keyCode === 27 && !this.props.viewState.shareModalIsVisible) {
```

Comment:
we have a problem here. If share panel on the top right hand corner is open and explorer window is open, esc key close the share panel, and further esc key press does nothing. I think this behaviour is problematic. Ideally the order should be: close catalog share -> close explorer window -> close map share panel

Reply:
Thanks for picking up on this too, the top-right-share-panel open + explorer window combination is fixed now but it also gets dismissed upon any further interaction inside the explorer window.

The end result is esc keys will always eventually bring you back to the map without modals open, but because we don't have any modal layer logic yet are you happy to leave it like this, then we raise an issue to create the new modal system post-mobx?